### PR TITLE
[AMLII-1410] Remove global dogstatsd server usage

### PR DESCRIPTION
**Pull request by rayz**
### What does this PR do?
This PR removes the usage of the global `var DSD server.Component` in the `dogstatsd/global.go` file. 

The `runner` that starts the JMXFetch process will now be explicitly innited and the `JMXFetch` struct now has a reference to the `DSD dogstatsdServer.Component`

### Motivation
Favor injecting the component over using a global

### Additional Notes
These changes should only affect JMXFetch as [this was the only place where the global `DSD` was still being used.](https://github.com/DataDog/datadog-agent/blob/070f8d1fb2afbe904ee933283cefca107350663b/pkg/jmxfetch/jmxfetch.go#L202)

### Describe how to test/QA your changes
1. Make sure JMXFetch still works
One way to do this:
  - Edit the misbehaving-server  docker-compose file with the agent image `datadog/agent-dev:raymond-zhao-remove-global-dogstatsd-reference-py3-jmx` https://github.com/DataDog/jmxfetch/blob/master/tools/misbehaving-jmx-server/docker-compose.yaml
  - Enable dogstatsd and uds by setting `DD_USE_DOGSTATSD` and `DD_DOGSTATSD_SOCKET`
  - `docker-compose up`
  - Make sure that JMXFetch is still working properly and metrics are being sent
2. Make sure jmx subcommand still works

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by gh123man — APPROVED** [review]
LGTM. 
I think we can remove the skip-qa tag for this - since you provided good QA steps, and I think this is worth validating at release time

File: cmd/agent/dogstatsd/global.go [hunk 1]
```diff
@@ -5,13 +5,3 @@
 
 // Package dogstatsd contains the dogstatsd subcommands.
 package dogstatsd
```
Comment (dustmop) [inline cmd/agent/dogstatsd/global.go:7]: Can we just remove this file entirely?

**Review by dustmop — APPROVED** [review]
Great to see a global usage removed! LGTM, with one suggestion

File: cmd/agent/dogstatsd/global.go (hunk 1 above)
Comment (rayz) [reply cmd/agent/dogstatsd/global.go:7]: Will do 👍

**Review by ogaca-dd — APPROVED** [review]
👍

**Pull request merged**